portfolio/src/main/java/com/google/sps/servlets/ListCommentsServlet.java

```diff
@@ -39,13 +40,23 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("Comment").addSort("timestamp", SortDirection.ASCENDING);
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
+    int nComments = getNCommentsChoice(request);
+
+    if (nComments == -1) {
+      response.setContentType("text/html");
+      response.getWriter().println("Please enter an integer between 1 and 3.");
```

Comment:
Why have two places where you print an error to the user (you have this message here but also print a message inside the getNCommentsChoice function.  I would recommend doing all the printing in a single place if possible.

Also, here is say to enter a value between 1 and 3 but in the function it doesn't check that the value is <=3.  Which behavior is correct?

Reply:
The message is incorrect, the correct behavior is to check for a positive integer.  

Also, I'm printing in two different instances, because inside the function I'm printing an error message to the terminal, while in line 52 I'm printing to the html file, so it can generate an alert.

Reply:
Ok, that makes sense.

I think the new message should say greater than or equal to 1 though, since 1 is a valid value.

Reply:
Ok, sounds good